osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/infra/messaging/CoreLogItemJsonMessageCreatorTest.java

```
@@ -0,0 +1,82 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.core.infra.messaging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CoreLogItemJsonMessageCreatorTest {
+  private Session session;
+
+  private CoreLogItemRequestMessage coreLogItemRequestMessage;
+  private CoreLogItemJsonMessageCreator jsonMessageCreator;
+
+  @BeforeEach
+  public void setUp() {
+    this.jsonMessageCreator = new CoreLogItemJsonMessageCreator();
+    this.coreLogItemRequestMessage = this.getCoreLogItemRequestMessage();
+    this.session = mock(Session.class);
+  }
+
+  @Test
+  void createJsonMessage() throws JMSException {
+    final TextMessage expectedTextMessage = mock(TextMessage.class);
+    when(this.session.createTextMessage(any())).thenReturn(expectedTextMessage);
+
+    final Message actualMessage =
+        this.jsonMessageCreator.getJsonMessage(this.coreLogItemRequestMessage, this.session);
+
+    assertThat(actualMessage).isSameAs(expectedTextMessage);
+  }
+
+  @Test
+  void jsonMessageIsEmptyWhenThereIsNoInputMessage() {
+    final Message actualMessage = this.jsonMessageCreator.getJsonMessage(null, this.session);
+
+    assertThat(actualMessage).isNull();
+  }
+
+  @Test
+  void jsonMessageIsEmptyWhenInputMessageHasNoValues() {
+    final Message actualMessage =
+        this.jsonMessageCreator.getJsonMessage(
+            new CoreLogItemRequestMessage(null, null, null), this.session);
+
+    assertThat(actualMessage).isNull();
+  }
```

Returning `null` as a `Message` when the input is missing entirely or is missing required values to be useful passes the problem. Failing fast is very likely a better way to deal with these scenarios.